src/estimation/include/iDynTree/Estimation/AttitudeQuaternionEKF.h

```diff
@@ -0,0 +1,332 @@
+     *
+     * Measurement model for accelerometer is given as,
+     * \f[ h_{acc}(\hat{x}_{k+1}) = \begin{bmatrix} 2(q_1q_3 - q_0q_2) \\ 2(q_2q_3 - q_0q_1) \\ q_0^2 - q_1^2 - q_2^2 + q_3^2 \end{bmatrix} \f]
+     * obtained from \f$ {^w}R_b^T e_3 \f$ of the assumed gravity direction.
+     *
+     * Measurement model for magnetometer measurement is given as,
+     * \f[ h_{mag}(\hat{x}_{k+1}) = atan2( 2(q_0q_3 + q_1q_2),1 - 2(q_2^2 + q_3^2) ) \f]
+     *
+     * The linearized system propogation and measurement model is obtained by computing Jacobins F and H with respect to the state.
+     *
+     * The zero mean, additive Gaussian noise can be set using the covariance matrices which will be used during predict and update steps.
+     *
+     * The propagateStates() method is called to set the input vector for the EKF, then ekfPredict() is called to propagate the state through the propagation function f()
+     * and propate the state covariance using the Jacobian F.
+     *
+     * The updateFilterWithMeasurements() is called to set the measurement vector for the EKF, and then ekfUpdate is used to correct the state estimate and its covariance
+     * using the measurement model function h() and the measurement Jacobian H.
+     *
+     * The usage of the QEKF should follow the decribed procedure below,
+     * - instantiate the filter
+     * - set parameters
+     * - call initializeFilter() (this is necessary for resizing the buffers, the user should call this method after setting parameters)
+     * - use setInternalState() to set initial state (The filter will throw an error, if this is not called atleast once, this enforces the user to set intial state)
+     * - Once initialized, the following filter methods can be run in a loop to get the orientation estimates,
+     *     - propagateStates() method to propagate the states and covariance
+     *     - updateFilterWithMeasurements() method to correct the predicted states and covariance
+     *     - getInternalState() or getOrientationEstimate*() methods to get the entire state estimate or only the attitude estimated in desired representation
+     *
+     * @warning calling the method useMagnetometerMeasurements() while the estimator is running, will reset the filter, reinitialize the filter to resize buffers
+     * and sets the previous estiamted state as the inital state.
+     * @note calling other set parameter methods does not reset the filter, since they are not associated with changing the output dimensions
+     *
+     */
+    class AttitudeQuaternionEKF : public IAttitudeEstimator,
+                                  public DiscreteExtendedKalmanFilter
+    {
+    public:
+        AttitudeQuaternionEKF();
+
+        /**
+         * @brief Get filter parameters as a struct.
+         * @param[out] params object of AttitudeQuaternionEKFParameters passed as reference
+         */
+        void getParams(AttitudeQuaternionEKFParameters& params) {params = m_params;}
```

Comment:
In Mahony, this is called `getParameters` . I think it would be good to stick with  one variant.